include/dlaf_c/eigensolver/eigensolver.h

```diff
@@ -0,0 +1,76 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <dlaf_c/desc.h>
+#include <dlaf_c/utils.h>
+
+#ifdef DLAF_WITH_SCALAPACK
+
+/// Eigensolver
+///
+/// The matrices $A$ and $Z$ are assumed to be distributed and in host memory.
+/// Moving to and from GPU memory is handled internally. The vector of eigenvalues $w$
+/// is assumed to be local (non-distributed).
+///
+/// The pika runtime is resumed when this function is called and suspended when the call
+/// terminates.
+///
+/// Submatrices are currently not supported, so @param n is the size of the full matrix $A$
+/// and @param ia, @param ja, @param iz, and @param jz need to be 1.
+///
+/// @param uplo indicates wheather the upper ('U') or lower ('L') triangular part of the global submatrix
+/// $A$ is referenced
+/// @param n order of the sumbatrix $A$ used in the computation
+/// @param a Local part of the global matrix $A$
+/// @param ia row index of the global matrix $A$ identifying the first row of the submatrix $A$, has to be 1
+/// @param ja column index of the global matrix $A$ identifying the firs column of the submatrix $A$, has to be 1
+/// @param desca ScaLAPACK array descriptor of the global matrix $A$
+/// @param w Local vector of eigenvalues (non-distributed)
+/// @param z Local part of the global eigenvectors matrix $Z$
+/// @param iz row index of the global matrix $Z$ identifying the first row of the submatrix $Z$, has to be 1
+/// @param jz column index of the global matrix $Z$ identifying the firs column of the submatrix $Z$, has to be 1
+/// @param descz ScaLAPACK array descriptor of the global eigenvectors matrix $Z$
+/// @param info 0 if the factorization completed normally
+DLAF_EXTERN_C void dlaf_pssyevd(char uplo, int n, float* a, int ia, int ja, int* desca, float* w,
+                                float* z, int iz, int jz, int* descz, int* info);
+
+/// \overload
```

Comment:
Just a reminder to check this in the doxygen output. I _think_ this should be `Something something see @ref dlaf_pssyevd, something this is for doubles.` This is not strictly an overload of another function (since it can't be in C).

Reply:
Good point. I looked into the Doxygen documentation (I haven't used it in a few yeas, hence I'm really rusty) and I think what we want is `@copydoc`.